File: alex-green-pantry-be/Pantry/src/main/java/net/yorksolutions/pantry/Models/Recipe.java
package net.yorksolutions.pantry.Models;

import jakarta.persistence.*;

import java.util.List;

@Entity
public class Recipe {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    public Long id;

    public String name;

    @Column(columnDefinition = "TEXT")
    public String imageURL;

    @ElementCollection
    public List<String> steps;

    public Long creatorId;

    @ManyToMany(cascade = CascadeType.MERGE)
    public List<Item> items;

    @ElementCollection
    public List<Float> itemWeights;

}

File: alex-green-pantry-fe/src/data/recipe.ts
import { Item } from "./item";

export class Recipe {
    public id: number | null = null;
    constructor(
        public name: String,
        public imageURL: String,
        public items: Item[],
        public steps: String[],
        public creatorId: number | null,
        public itemWeights: number[]
    ){}
}